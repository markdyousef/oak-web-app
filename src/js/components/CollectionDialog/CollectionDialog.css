:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
    --line-gray-color: #e5e5e5;
}

.modal {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    z-index: 2000;
    background-color: rgba(0, 0, 0, 0.4);
    display: flex;
    align-items: center;
    justify-content: center;
}

.container {
    width: 400px;
    height: 240px;
    background-color: #fff;
    border-radius: 3px;
    border: 1px solid #ddd;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
}

.title {
    width: 100%;
    height: 60px;
    border-bottom: 1px solid var(--line-gray-color);
    display: flex;
    align-items: center;
    padding: 20px;
    font-size: 16px;
    font-weight: bold;
}

.main {
    padding: 5px 20px;
    width: 100%;
}

.buttons {
    padding: 15px 15px;
    width: 100%;
    display: flex;
    justify-content: space-between;
}
