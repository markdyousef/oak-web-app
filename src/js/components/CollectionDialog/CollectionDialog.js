import React, { Component, PropTypes } from 'react';
import Input from '../shared/Input';
import Button from '../shared/Button';

import css from './CollectionDialog.css';

class CollectionDialog extends Component {
    static propTypes = {
        close: PropTypes.func.isRequired
    }
    static defaultProps = {}
    constructor() {
        super();
        this.state = {
            name: ''
        };
    }
    render() {
        const { close } = this.props;
        const { name } = this.state;
        return (
            <div className={css.modal}>
                <div className={css.container}>
                    <div className={css.title}>
                        <h1>Add/Edit collection: </h1>
                    </div>
                    <div className={css.main}>
                        <Input
                            title="NAME"
                            value={name}
                            onChange={value => this.setState({ name: value })}
                            placeholder="Name"
                        />
                    </div>
                    <div className={css.buttons}>
                        <Button
                            onClick={close}
                            text="Cancel"
                        />
                        <Button
                            onClick={close}
                            text="Save"
                            type="primary"
                        />
                    </div>
                </div>
            </div>
        );
    }
}

export default CollectionDialog;
