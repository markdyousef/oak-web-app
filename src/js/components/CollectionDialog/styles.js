import styled from 'styled-components';
import colors from '../../styles/colors';

export const Container = styled.section`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    z-index: 9999;
    background-color: rgba(19, 21, 23, 0.5);
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Header = styled.div`
    width: 100%;
    height: 60px;
    border-bottom: 1px solid ${colors.lightGrey};
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 20px;
    & h1 {
        font-size: 20px;
        font-weight: bold;
    }
`;

export const Close = styled.button`
    cursor: pointer;
    background-color: transparent;
    border: 0;
    font-size: 45px;
    font-weight: 100;
    margin-top: -8px;
    margin-right: -10px;
    color: rgba(19, 21, 23, 0.5);
    &:hover {
        color: rgba(19, 21, 23, 1);
    }
`;

export const Main = styled.div`
    padding: 20px;
    width: 100%;
`;

export const Intro = styled.div`
    font-size: 16px;
    color: #131517;
    line-height: 1.35;
`;

export const InputWrapper = styled.div`
    padding-top: 24px;
`;

export const InputLabels = styled.div`
    font-size: 14px;
    font-weight: bold;
    color: rgba(19, 21, 23, 0.75);
    padding-bottom: 4px;
`;

export const InputName = styled.div`
    font-size: 13px;
    font-weight: normal;
    color: rgba(19, 21, 23, 0.5);
    padding-top: 6px;
`;

export const Upload = styled.div`
    height: 210px;
    border: 1px solid ${colors.lightGrey};
    border-radius: 2px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 20px;
    background-color: ${colors.lightGrey};
    cursor: pointer;
    & img {
        height: 100%;
        width: 100%;
        border-radius: 3px;
    }
`;

export const Buttons = styled.div`
    border-top: 1px solid #e5e5e5;
    padding: 20px;
    width: 100%;
    display: flex;
    justify-content: space-between;
`;
