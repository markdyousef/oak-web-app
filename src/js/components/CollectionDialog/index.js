import CollectionDialog from './CollectionDialog';

export default CollectionDialog;
