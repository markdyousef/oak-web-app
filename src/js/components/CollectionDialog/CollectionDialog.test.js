import React from 'react';
import renderer from 'react-test-renderer';
import CollectionDialog from './CollectionDialog';

it('renders correctly', () => {
    const props = {
        close: () => {}
    }
    const component = renderer.create(
        <CollectionDialog {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
