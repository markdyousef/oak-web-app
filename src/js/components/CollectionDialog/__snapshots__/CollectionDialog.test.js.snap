exports[`test renders correctly 1`] = `
<section
  className="bGtZFR">
  <div
    className="hsANYj">
    <div
      className="jqVAxc">
      <h1>
        Add/Edit collection: 
      </h1>
      <button
        className="gxoqIZ"
        onClick={[Function]}>
        Ã—
      </button>
    </div>
    <div
      className="hbuMgW">
      <div
        className="ztAri"
        onClick={[Function]}>
        <img
          alt="cover"
          src="test-file-stub" />
        <input
          accept="image/*"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file" />
      </div>
      <div>
        <h5
          className="dRMKoo" />
        <input
          className="bwXyts"
          onChange={[Function]}
          onKeyDown={null}
          placeholder="Name your collection"
          type="text"
          value="" />
      </div>
      <div>
        <h5
          className="eKaiQM">
          
        </h5>
        <textarea
          className="cVtAGf"
          onChange={[Function]}
          placeholder="Add a description..."
          value="" />
      </div>
    </div>
    <div
      className="kKvTgr">
      <button
        className="jpCUHi"
        onClick={[Function]}
        type={undefined}>
        Cancel
      </button>
      <button
        className="hQjvYS"
        onClick={[Function]}
        type="primary">
        Save
      </button>
    </div>
  </div>
</section>
`;
